tests/IntegrationTest/Definitions/WildcardDefinitionsTest.php

```diff
@@ -52,6 +67,24 @@ public function test_wildcards_autowire(ContainerBuilder $builder)
         $object = $container->get(Interface1::class);
         $this->assertInstanceOf(Implementation1::class, $object);
 
+        self::assertEntryIsNotCompiled($container, 'DI\Test\IntegrationTest\*\Interface*');
+        self::assertEntryIsNotCompiled($container, Interface1::class);
+        self::assertEntryIsNotCompiled($container, Interface2::class);
+    }
```

Comment:
I've added a simpler version of your test: this test passes on the current master. I think this is better to have a simple test (easy to grasp) which covers the base of the feature + your test that is more advanced and covers the bug you were facing.